sklearn/preprocessing/_function_transformer.py

```diff
@@ -67,16 +76,36 @@ class FunctionTransformer(BaseEstimator, TransformerMixin):
 
     """
     def __init__(self, func=None, inverse_func=None, validate=True,
-                 accept_sparse=False, pass_y='deprecated',
+                 accept_sparse=False, pass_y='deprecated', check_inverse=True,
                  kw_args=None, inv_kw_args=None):
         self.func = func
         self.inverse_func = inverse_func
         self.validate = validate
         self.accept_sparse = accept_sparse
         self.pass_y = pass_y
+        self.check_inverse = check_inverse
         self.kw_args = kw_args
         self.inv_kw_args = inv_kw_args
 
+    def _check_inverse_transform(self, X):
+        """Check that func and inverse_func are the inverse."""
+        idx_selected = slice(None, None, max(1, X.shape[0] // 100))
+        try:
+            assert_allclose_dense_sparse(
+                X[idx_selected],
+                self.inverse_transform(self.transform(X[idx_selected])))
+        except AssertionError:
+            # TODO: an error will be raised from 0.22
+            # raise ValueError("The provided functions are not strictly"
+            #                  " inverse of each other. If you are sure you"
+            #                  " want to proceed regardless, set"
+            #                  " 'check_inverse=False'.")
+            warnings.warn("The provided functions are not strictly"
+                          " inverse of each other. If you are sure you"
+                          " want to proceed regardless, set"
+                          " 'check_inverse=False'. This warning will turn to"
+                          " an error from 0.22", DeprecationWarning)
```

Comment:
It should not be a DeprecationWarning. It's a UserWarning, I think, or perhaps a RuntimeWarning.

We could just allow the user to use a warning filter to make it raise an error (particularly if we define a new warning class), rather than changing behaviour in two versions. I think that's okay in the transformed target case too...